legal-api/src/legal_api/reports/report.py

```diff
@@ -171,12 +175,18 @@ def _get_template_data(self):  # pylint: disable=too-many-branches
                 self._format_with_diff_data(filing)
 
             # name change from named company to numbered company case
-            if self._report_key == 'certificateOfNameChange' and 'legalName' not in filing['alteration']['nameRequest']:
+            if self._report_key in ('certificateOfNameChange', 'alterationNotice') and 'nameRequest' in \
+                    filing['alteration'] and 'legalName' not in filing['alteration']['nameRequest']:
                 versioned_business = \
                     VersionedBusinessDetailsService.get_business_revision_after_filing(self._filing.id,
                                                                                        self._business.id)
                 filing['alteration']['nameRequest']['legalName'] = versioned_business['legalName']
 
+            if self._report_key == 'alterationNotice':
+                versioned_business = VersionedBusinessDetailsService.get_business_revision_before_filing(
```

Comment:
re-using the same name as above enables future errors, especially as the intent of the variable is different.
please rename

Reply:
Fixed.